core/src/main/java/jenkins/model/NewViewLink.java

```diff
@@ -44,7 +44,7 @@ public String getDisplayName() {
 
             @Override
             public String getUrlName() {
-                return Jenkins.get().getRootUrl() + URL_NAME;
+                return URL_NAME;
```

Comment:
```suggestion
                return "/" + URL_NAME;
```

The relative view will not work within an already existing view as mentioned by https://github.com/jenkinsci/jenkins/pull/3178. It's the case for regular "List View" (Core). For certain view, like Nested View (plugin), the feature is expected and it adds itself the action button to create new View inside the nested view, resulting in duplicate (no change with your PR).

That fix was incorrect for the proxy URL that you are trying to solve, but with your solution you are going to introduce previous problem back.

Multiple ways to improve the situation without coming back with regression:
1) Use "absolute" relative link, meaning, it's relative to the contextPath, but absolute *inside* Jenkins. If you are in a folder or in a view, the action will propose a new view "globally". This is a bit counter intuitive in terms of UX. (that's the suggestion, the easiest way but not perfect one)
2) Add support to create view from all the views (could be tricky)
3) Do not display the generic new View, if the current ancestor is a view.

Concerning the point 3, perhaps something to do on this line: [NewViewLink.java#L25](https://github.com/jenkinsci/jenkins/blob/0e62b68012282648377cfac4fe6d01956d7c1f11/core/src/main/java/jenkins/model/NewViewLink.java#L25). At this point, if you are in a `ListView`, we should not display the link as ListView does not support nested view. If we are in a folder or in Jenkins home page, the v is an instance of `AllView`, and the link will be to `AbstractFolder/newView.jelly` or `Jenkins/newView.jelly` respectively, leading once posted, to `AbstractFolder#doCreateNewView` or `Jenkins#doCreateNewView`. There are perhaps mechanisms that I am not aware to inform that "you" can accept new view creation or similar stuff.

I would recommend to just go with proposal 1 and in a follow-up PR to look at other solutions.